builtin/providers/aws/resource_aws_instance.go

```diff
@@ -760,6 +760,10 @@ func fetchRootDeviceName(ami string, conn *ec2.EC2) (*string, error) {
 		rootDeviceName = image.BlockDeviceMappings[0].DeviceName
 	}
 
+	if rootDeviceName == nil {
+		return nil, fmt.Errorf("[WARN] Error finding Root Device Name for AMI (%s)", ami)
```

Comment:
@phinze did you see my updates here? Hopefully they made sense, let's sync up and see what we need to do